src/ripple/nodestore/impl/DatabaseShardImp.cpp

```diff
@@ -748,6 +921,94 @@ DatabaseShardImp::selectCache(std::uint32_t seq)
     return cache;
 }
 
+bool
+DatabaseShardImp::importCopy(std::unique_lock<std::mutex>&,
```

Comment:
This looks a lot like `DatabaseShardImp::copyLedger`. Are they too different to unify?

Reply:
This one lacks some checks that are unnecessary here. It also doesn't lock and make changes to the `complete_` and `incomplete_` members. Its possible to unify but not sure if it warrants the effort.

Reply:
I forgot to mention this version does not pollute the shard's caches, a performance improvement.